utils/usePopper.ts

```diff
@@ -26,16 +38,17 @@ export function usePopper(options) {
     eventsEnabled,
     positionFixed
   } = props;
-  const [state, setState] = useState(() => ({
+
+  const [state, setState] = useState<IUsePopperState>(() => ({
     data: undefined,
     placement: undefined
   }));
-  const popperInstance = useRef();
+  const popperInstanceRef = useRef<PopperJS | null>();
 
   useEffect(() => {
     if (referenceRef.current && popperRef.current) {
-      popperInstance.current = new PopperJS(referenceRef.current, popperRef.current, {
-        placement: popperPlacement,
+      popperInstanceRef.current = new PopperJS(referenceRef.current, popperRef.current, {
+        placement: popperPlacement as Placement,
```

Comment:
Isn't `popperPlacement` already types as `Placement`? This may not be necessary

Reply:
The `popperPlacement` value that is passed into `usePopper` as an argument and is inferred to be a `string`. I do not think it is already typed as `Placement`.

Without casting this value as `Placement`, TypeScript complains with: 

![Screen Shot 2019-10-31 at 10 17 58 AM](https://user-images.githubusercontent.com/1811365/67970132-bda0b380-fbc7-11e9-884b-075bda4a13ae.png)

Reply:
Maybe we should type the `placement` string as a `Placement` then! Since those are the only valid variables we can use that type here to improve the specificity.

Reply:
> Maybe we should type the placement string as a Placement then!

This was the approach I went with 💯, thanks!